### PR TITLE
fix: retrieve deployed collection names in AWS

### PR DESCRIPTION
**Pull request by jyecusch**
More work required in future, but tracked in a separate issue.

### CODE REVIEW

**Comment by jyecusch** [comment]
depends on https://github.com/nitrictech/cli/pull/129

File: pkg/plugins/document/dynamodb/dynamodb.go [hunk 1]
```diff
@@ -556,20 +580,37 @@ func isBetweenEnd(index int, exps []document.QueryExpression) bool {
...
 	coll := collection
 	for coll.Parent != nil {
 		coll = *coll.Parent.Collection
 	}
 
-	return aws.String(coll.Name)
+	// TODO: The following method for determining the deployment specific table name from the nitric name is unreliable.
+	//	a new design is in process and will replace this method for locating tables.
+	out, err := s.client.ListTables(&dynamodb.ListTablesInput{})
+
```
Comment (medgar-nitric) [inline pkg/plugins/document/dynamodb/dynamodb.go:592]: I think we should be caching these table name look results in memory. We can have a retry if the table value is not found, was not found to handle any deployment timing issues.

File: pkg/plugins/document/dynamodb/dynamodb.go [hunk 2]
```diff
@@ -556,20 +580,37 @@ func isBetweenEnd(index int, exps []document.QueryExpression) bool {
...
+	if err != nil {
+		return nil, fmt.Errorf("encountered an error retrieving the bucket list: %v", err)
```
Comment (medgar-nitric) [inline pkg/plugins/document/dynamodb/dynamodb.go:594]: Change message to: `encountered an error retrieving the table list: %v`

**Review by medgar-nitric — CHANGES_REQUESTED** [review]

File: pkg/plugins/document/dynamodb/dynamodb.go (hunk 1 above)
Comment (jyecusch) [reply pkg/plugins/document/dynamodb/dynamodb.go:606]: Good suggestion, I've added some basic caching now

**Review by tjholm — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -18,11 +18,12 @@ The Membrane is at the heart of the solution. Nitric applications communicate wi
   - Queues
   - Key/Value Store
```
Comment (medgar-nitric) [inline README.md:19]: Should remove Key/Value Store

Comment (jyecusch) [reply README.md]: Done

**Review by medgar-nitric — APPROVED** [review]

**Pull request merged**